### PR TITLE
feat: add tests for AddAndMove and AddBuildingBlocks to canvas

### LINKED ISSUE #34146: [Task]: create test case for dragging building block to canvas with reflow
### SubTasks

Adds test for this PR - #34143

### PR DESCRIPTION
**Pull request by jacquesikot**
## Description
Ad unit tests for adding and moving building blocks saga.

Fixes #34146

## Automation

/ok-to-test tags="@tag.Sanity"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9494392536>
> Commit: c6b9b1b679f284458cc6500729408ddb015adb62
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9494392536&attempt=1" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **Tests**
  - Added comprehensive tests for adding and moving building blocks on the canvas.
  - Updated import paths for consistency in test files.
  - Introduced new test suites for sagas handling building block operations.

- **New Fixtures**
  - Added mock data and constants for testing building block functionalities.

### CODE REVIEW

File: app/client/src/sagas/BuildingBlockSagas/tests/AddAndMoveBuildingBlockToCanvasSaga.test.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+import { skeletonWidget } from "./fixtures";
+
+type GeneratorType = Generator<
+  CallEffect | SelectEffect | PutEffect,
+  void,
+  unknown
+>;
+
+describe("addAndMoveBuildingBlockToCanvasSaga", () => {
+  const actionPayload: ReduxAction<{
```
Comment (rahulbarwal) [inline app/client/src/sagas/BuildingBlockSagas/tests/AddAndMoveBuildingBlockToCanvasSaga.test.ts:25]: Nit: we can move this to fixture file.

File: app/client/src/sagas/BuildingBlockSagas/tests/AddBuildingBlockToCanvasSaga.test.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import { skeletonWidget } from "./fixtures";
+
+type GeneratorType = Generator<
+  CallEffect | SelectEffect | PutEffect,
+  void,
+  unknown
+>;
+
+describe("addBuildingBlockToCanvasSaga", () => {
+  const addEntityAction: ReduxAction<WidgetAddChild> = {
```
Comment (rahulbarwal) [inline app/client/src/sagas/BuildingBlockSagas/tests/AddBuildingBlockToCanvasSaga.test.ts:23]: Same as above: move to fixture file.

**Review by rahulbarwal — CHANGES_REQUESTED** [review]
Requesting minor changes. Over all looks good.

File: app/client/src/sagas/BuildingBlockSagas/tests/AddBuildingBlockToCanvasSaga.test.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        loadBuildingBlocksIntoApplication,
+        addEntityAction.payload,
+        skeletonWidget.widgetId,
+      ),
+    );
+
+    // Complete the generator
+    result = generator.next();
+    expect(result.done).toBe(true);
+  });
```
[the remark this answers is not in the document]
Comment (jacquesikot) [reply app/client/src/sagas/BuildingBlockSagas/tests/AddBuildingBlockToCanvasSaga.test.ts:112]: @coderabbitai can you give me an example of this intermediate steps in the code?

**Review by rahulbarwal — APPROVED** [review]

**Pull request merged**